src/App.tsx

```diff
@@ -14,11 +14,63 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    selectedGoods: ['Jam'],
+  };
+
+  getButtonWith = (text: string, setState: string[], addclassName?: string) => {
```

Comment:
this method is not good for understanding and there is no necessity to do it. Better create method where you need to update state and pass needed parameters onClick, but it should be 3 different buttons

Reply:
ok